src/widgets/NewExpiryDateInput.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/* eslint-disable react/forbid-prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { View, TouchableWithoutFeedback, Text, TextInput } from 'react-native';
+import { useExpiryDateMask } from '../hooks/useExpiryDateMask';
+import Cell from './DataTable/Cell';
+import { dataTableColors, newDataTableStyles } from '../globalStyles/index';
+import { getAdjustedStyle } from './DataTable/utilities';
+import { parseExpiryDate, formatExpiryDate } from '../utilities';
+
+/**
+ * Renders an expiry date cell, managing its own state and not submitting
+ * a date to be committed to the underlying model until a valid date has
+ * been submitted.
+ *
+ * Similar to EditableCell, this component renders a Cell when disabled,
+ * a Text wrapped in a TouchableComponent when un-focused and a TextInput
+ * when focused - A workaround for react-native crashes when large numbers
+ * of TextInput components are rendered.
+ *
+ * @param {string|number} value The value to render in cell
+ * @param {string|number} rowKey Unique key associated to row cell is in
+ * @param {string|number} columnKey Unique key associated to column cell is in
+ * @param {bool} disabled If `true` will render a plain Cell element with no interaction
+ * @param {bool} isFocused If `false` will TouchableOpacity that dispatches a focusAction
+ *                         when pressed. When `true` will render a TextInput with focus
+ * @param {func} editAction Action creator for handling editing of this cell.
+ *                          `(newValue, rowKey, columnKey) => {...}`
+ * @param {func} focusAction Action creator for handling focusing of this cell.
+ *                          `(rowKey, columnKey) => {...}`
+ * @param {func} focusNextAction Action creator for handling focusing of this cell.
+ *                          `(rowKey, columnKey) => {...}`
+ * @param {func}  dispatch Reducer dispatch callback for handling actions
+ * @param {Number}  width Optional flex property to inject into styles.
+ * @param {Bool}  isLastCell Indicator for if this cell is the last
+ *                                   in a row. Removing the borderRight if true.
+ * @param {String}  placeholder String to display when the cell is empty.
+ * @
+ *
+ */
+export const NewExpiryDateInput = React.memo(
+  ({
+    value,
+    rowKey,
+    columnKey,
+    isDisabled,
+    isFocused,
+    editAction,
+    focusAction,
+    focusNextAction,
+    dispatch,
+    isLastCell,
+    width,
+    debug,
+    placeholder,
+  }) => {
+    if (debug) console.log(`- ExpiryTextInputCell: ${value}`);
+
+    // Customhook managing the editing of an expiry date to stay valid.
+    const [expiryDate, finalseExpiryDate, setExpiryDate] = useExpiryDateMask(
```

Comment:
dropped an `i` in `finalseExpiryDate`

Reply:
I think typically the return from the custom hook file should match the usage?

Reply:
first - 09652193d7dec278fdebff2c81685606f93b4a17

second - 5d5fabe5697de02f7f593b4616d2fa1e884017be